src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -975,7 +1005,10 @@ private void InsertToStack(CursorState newPageState)
         }
 
         private void DefragPage()
-        {
+        {                     
+            // TODO: The process of defrag could be the perfect place to upgrade a dictionary, as we are certainly going to be writing this new
+            //       page anyways; if that is the case, why not do it with the new dictionary.
+
```

Comment:
Issue here is that this will usually be triggered on writes / deletes / write.
So we may not have that many items in the page. Also, that can cause us to have many dics.

Reply:
The idea for that is the following:
- Whenever we hit this, we are gonna having to spend the IO for the page (which is what it is actually expensive). 
- The runtime cost of having to 'accommodate' different dictionaries grows linearly with the depth of the tree (when we have a key with 1 encoding and searching a page with a different one). 
So it is actually a good idea to upgrade a page when we are going to be spending the time to write it.

Reply:
There is no I/O on the page that is special for defrag.
_Any_ change to the pager requires us to do a full read / write, basically.

Reply:
Exactly, if the page requires to be written; we should take the time and upgrade it.

Reply:
There is no distinction between normal write and defrag - both have ~ same cost.

Reply:
The write of a page does indeed happen in both cases, I am thinking in terms of the cost of decoding + encoding a page. Which shouldn't be high but would still cause allegedly small fluctuations in runtime cost. Since those happen at write time, I don't think we should care much about them.